client/my-sites/upgrades/domain-management/edit/card/header/index.jsx

```diff
@@ -21,16 +24,15 @@ const Header = React.createClass( {
 	},
 
 	render() {
-		const domain = this.props.domain;
+		const { domain, isDomainOnly } = this.props;
 
-		if ( ! domain ) {
+		if ( ! domain || isDomainOnly ) {
```

Comment:
Instead of hiding this header altogether, I think we should just be hiding the primary flag for domain-only sites.

There is already a condition for hiding it for domains that are not the primary domain [here](https://github.com/Automattic/wp-calypso/blob/fcf789ebb43b7d8fe9a912d8fa82d5bee66573b5/client/my-sites/upgrades/domain-management/components/domain/primary-flag.jsx#L17), so we can potentially add to that.